### PR TITLE
feat(ecs): add external network modes to ExternalTaskDefinition and TaskDefinition

### PR DESCRIPTION
**Pull request by beezly**
Adds support for Host and None network modes when creating Task Definitions with External compatibility.

See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere.html  

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
Hey @neilkuan @clarencetw @dwchiang @kimisme9386 can you take a first look?

File: packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
 import * as ssm from '@aws-cdk/aws-ssm';
 import * as cdk from '@aws-cdk/core';
 import * as ecs from '../../lib';
+import { NetworkMode } from '../../lib';
```
Comment (neilkuan) [inline packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts:10]:
```suggestion
```

File: packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts [hunk 2]
```diff
@@ -37,6 +38,7 @@ describe('external task definition', () => {
           ),
         }),
         family: 'ecs-tasks',
+        networkMode: NetworkMode.HOST,
```
Comment (neilkuan) [inline packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts:41]:
```suggestion
        networkMode: ecs.NetworkMode.HOST,
```

File: packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts [hunk 3]
```diff
@@ -66,6 +68,18 @@ describe('external task definition', () => {
...
     });
 
+    test('error when an invalid networkmode is set', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // THEN
+      expect(() => {
+        new ecs.ExternalTaskDefinition(stack, 'ExternalTaskDef', {
+          networkMode: NetworkMode.AWS_VPC,
```
Comment (neilkuan) [inline packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts:78]:
```suggestion
          networkMode: ecs.NetworkMode.AWS_VPC,
```

File: packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts [hunk 4]
```diff
@@ -401,8 +401,8 @@ export class TaskDefinition extends TaskDefinitionBase {
       throw new Error('Cannot use inference accelerators on tasks that run on Fargate');
     }
 
-    if (this.isExternalCompatible && this.networkMode !== NetworkMode.BRIDGE) {
-      throw new Error(`External tasks can only have Bridge network mode, got: ${this.networkMode}`);
+    if (this.isExternalCompatible && ![NetworkMode.BRIDGE, NetworkMode.HOST, NetworkMode.NONE].includes(this.networkMode)) {
```
[the remark this answers is not in the document]
Comment (beezly) [reply packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts:404]: Hi @neilkuan. The behaviour is described in https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere.html:

> The tasks you run on your external instances must use the bridge, host, or none network modes. The awsvpc network mode isn't supported.

File: packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts (hunk 1 above)
Comment (beezly) [reply packages/@aws-cdk/aws-ecs/test/external/external-task-definition.test.ts]: @neilkuan I've merged all these into a new commit.

**Comment by beezly** [comment]
@pahud is anything else needed for this - I also have a branch building against the v2 code. https://github.com/beezly/aws-cdk/tree/v2_network_modes

File: packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts [hunk 5]
```diff
@@ -15,7 +15,14 @@ import {
  * The properties for a task definition run on an External cluster.
  */
 export interface ExternalTaskDefinitionProps extends CommonTaskDefinitionProps {
-
+  /**
+   * The networking mode to use for the containers in the task.
+   *
+   * With ECS Anywhere, supported modes are bridge, host and none.
+   *
+   * @default - NetworkMode.Bridge
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts:23]:
```suggestion
   * @default NetworkMode.BRIDGE
```

File: packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts [hunk 6]
```diff
@@ -59,20 +65,27 @@ export class ExternalTaskDefinition extends TaskDefinition implements IExternalT
...
+      networkMode: attrs.networkMode,
       taskRole: attrs.taskRole,
     });
   }
 
+  /**
+   * The networking mode to use for the containers in the task.
+   */
+  public readonly networkMode: NetworkMode;
+
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts:77]: This public property is not used anywhere, so let's not add it until there is a requirement for it.

File: packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts [hunk 7]
```diff
@@ -59,20 +65,27 @@ export class ExternalTaskDefinition extends TaskDefinition implements IExternalT
...
    */
   constructor(scope: Construct, id: string, props: ExternalTaskDefinitionProps = {}) {
     super(scope, id, {
       ...props,
       compatibility: Compatibility.EXTERNAL,
-      networkMode: NetworkMode.BRIDGE,
+      networkMode: (props.networkMode ?? NetworkMode.BRIDGE),
     });
+
+    this.networkMode = props.networkMode ?? NetworkMode.BRIDGE;
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts:88]:
```suggestion
```

**Review by madeline-k — CHANGES_REQUESTED** [review]
Thanks for opening this PR, @beezly! Just a few minor comments, and then it is ready to go.

File: packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts (hunk 6 above)
Comment (madeline-k) [reply packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts:77]: Actually this already exists in the base class, TaskDefinition. It should still be removed from here.

File: packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts (hunk 7 above)
Comment (madeline-k) [reply packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts:88]: This is already done by passing in `props.networkMode ?? NetworkMode.BRIDGE` to the super constructor.

**Review by madeline-k — DISMISSED** [review]
Approving

**Comment by beezly** [comment]
@madeline-k looks like this didn't merge but I'm not sure why. Are you able to give me a hint?

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**